packages/transactions/src/serializers/__tests__/transaction-version-test.ts

```diff
@@ -0,0 +1,22 @@
+import { TransactionVersion } from '../../types';
+import { transactionVersionHeader } from '../transaction-version';
+
+const VERSION_FLAG_MASK = 0x80;
+
+describe('Transaction version serializer', () => {
+    it('serializes no data when the version is `legacy`', () => {
+        expect(transactionVersionHeader.serialize('legacy')).toEqual(new Uint8Array());
+    });
+    it.each(
+        // Versions 0–127
+        [...Array(128).keys()].map(version => [version | VERSION_FLAG_MASK, version as TransactionVersion])
+    )('serializes to `%s` when the version is `%s`', (expected, version) => {
+        expect(transactionVersionHeader.serialize(version)).toEqual(new Uint8Array([expected]));
+    });
+    it.each([-1 as TransactionVersion, 128 as TransactionVersion])(
+        'throws when passed the out-of-range version `%s`',
+        version => {
+            expect(() => transactionVersionHeader.serialize(version)).toThrow();
+        }
+    );
```

Comment:
Should we add tests for deserialize too? To make sure value -> serialize -> deserialize -> value works correctly